rs/registry/canister/src/registry.rs

```diff
@@ -511,6 +516,27 @@ impl Registry {
             }
         }
     }
+
+    /// Ensure that timestamps of mutations are in seconds.
+    fn ensure_mutation_timestamp_unit_seconds(
+        mutation_request: &mut HighCapacityRegistryAtomicMutateRequest,
+    ) {
+        // We use the beginning of the year in nanoseconds to determine if the
+        // current mutation has a timestamp in nanoseconds or in seconds.
+        //
+        // We know that the faulty mutations that were created, got created
+        // after this timestamp (in nanoseconds) and will convert them back
+        // to seconds.
+        //
+        // If the timestamp was already once converted back, it will not be
+        // done again.
+        const BEGINNING_OF_2025_NANOSECONDS: u64 = 1_735_693_261_000_000_000;
+
+        if mutation_request.timestamp_seconds > BEGINNING_OF_2025_NANOSECONDS {
```

Comment:
No change.

Yes, this is what I had in mind for how to make sure that we do not multiply scale down the same value.

Ofc, we would eventually get rid of this, but just in case we neglect to do that soon enough, it is safer to have this.